syft/frameworks/crypten/worker_support.py

```diff
@@ -0,0 +1,68 @@
+from typing import Dict
+import types
+
+from syft.workers.base import BaseWorker
+
+from syft.messaging.message import ObjectMessage
+from syft.messaging.message import CryptenInit
+
+from syft.frameworks.crypten import run_party
+
+
+def get_worker_from_rank(worker: BaseWorker, rank: int) -> BaseWorker:
+    assert hasattr(worker, "rank_to_worker_id"), "First need to call run_crypten_party"
+    return worker._get_worker_based_on_id(worker.rank_to_worker_id[rank])
+
+
+def _set_rank_to_worker_id(worker: BaseWorker, rank_to_worker_id: Dict[int, int]) -> None:
+    worker.rank_to_worker_id = rank_to_worker_id
+
+
+def run_crypten_party(worker: BaseWorker, message: tuple) -> ObjectMessage:
+    """Run crypten party according to the information received.
+
+        Args:
+            message (CryptenInit): should contain the rank_to_worker_id, world_size, master_addr and master_port.
+
+        Returns:
+            An ObjectMessage containing the return value of the crypten function computed.
+        """
+
+    worker.rank_to_worker_id, world_size, master_addr, master_port = message.crypten_context
+
+    plans = worker.search("crypten_plan")
+    assert len(plans) == 1
+
+    plan = plans[0].get()
+
+    rank = None
+    for r, worker_id in worker.rank_to_worker_id.items():
+        if worker_id == worker.id:
+            rank = r
+            break
+
+    assert rank != None
+
+    return_value = run_party(plan, rank, world_size, master_addr, master_port, (), {})
+    return ObjectMessage(return_value)
+
+
+def add_support_to_worker(worker):
+    worker._message_router[CryptenInit] = types.MethodType(run_crypten_party, worker)
```

Comment:
What is the role of `types.MethodType` ?

Reply:
Ah ok :) https://stackoverflow.com/questions/37455426/advantages-of-using-methodtype-in-python